### ISSUE TITLE
Hi how to train your code with different pre-trained models

### ISSUE THREAD

**Issue by ssbilakeri** [huanghoujing/person-reid-triplet-loss-baseline#43]
How to train with different pre-trained models.
please reply.

**Comment by huanghoujing** [comment]
Hi, to use another backbone.

- First, you create another file, e.g. `resnext.py` under directory `tri_loss/model`. The content is the original `resnext.py` provided by pytorch.
- Then, modify the `forward` function of `resnext`, removing unnecessary operation after this line `x = self.layer4(x)`, so that it returns the result of `layer4`.
- Finally, you can use `from .resnext import resnext50` in `tri_loss/model/Model.py`, and then replace this line
```python
self.base = resnet50(pretrained=True, last_conv_stride=last_conv_stride)
```
with
```python
self.base = resnext50(pretrained=True)
```

If you would like to reduce the last convolutional stride of the backbone, you can modify it yourself in `resnext.py` accordingly.

**Comment by ssbilakeri** [comment]
Thank you very much for your clear explanation.

I want to check with different networks like Efficient net, Inception.
Where do I get pretrained weight for these networks.

Can you suggest which pretrained network is preferred for reidentification.
Thank you.

**Comment by ssbilakeri** [comment]
Can I get anystrong literature to choose best pre-trained model for people
reidentification. Kindly suggest
Thank you.

**Comment by huanghoujing** [comment]
Sorry for the late response. If you want to try on many different backbones or understand which one is better, as well as some training tricks, you can read the paper ***FastReID: A Pytorch Toolbox for General Instance Re-identification*** and the accompanying code https://github.com/JDAI-CV/fast-reid.
